components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/bus/CryptoFlowOpsBusProcessor.kt

```
@@ -26,12 +32,14 @@ import net.corda.utilities.MDC_FLOW_ID
 import net.corda.utilities.trace
 import net.corda.utilities.withMDC
 import org.slf4j.LoggerFactory
+import java.nio.ByteBuffer
 import java.time.Instant
+import kotlin.math.sign
 
 class CryptoFlowOpsBusProcessor(
-    private val cryptoOpsClient: CryptoOpsProxyClient,
+    private val signingService: SigningServiceImpl,
```

It's a bit weird to use the impl instead of the interface here.  Kind of makes you wonder _why_ the interface?

---

I do often wonder why we have the interface. But in this case it is deliberate, the reason is that I want to use a variant of an entry point that does not log errors and backtraces to the console.

Maybe that should be in the interface but it doesn't really need to be.

---

Thinking more about this, @fowlerrr  is right; so I want to get a group decision on whether suppressing the error logging is important enough to justify the silent parameter, then either add it in to the interface or delete it.

---

I don't think you need `SigningServiceImpl` for that, though.  You can just use `SigningService` here.

Regarding `onNextSilent`, I don't really like it.  I get the idea, and I understand that logging errors during a test can be difficult to parse, but you're creating two different (albeit nearly identical) paths: one for test and one for production.  I don't like that as a rule and this sets a precedent I'd rather avoid.

---

The alternatives would be:
1.  Add a `silent` flag to `DurableProcessor.onNext`, in the interface?
2. Some kind of contextual stuff, e.g. thread local storage
3. give up and just log the error and backtrace

---

I gave up. The right solution might be to dynamically disable logging at the right time, as discussed here https://stackoverflow.com/questions/28807545/slf4j-or-logback-turn-off-logging-for-1-unit-test-or-1-thread but that requires a non-public API and maybe a new slf4j.